src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -482,6 +482,9 @@ private void copyDirectory(UnixPath source,
         } catch (UnixException x) {
             if (x.errno() == EEXIST && flags.replaceExisting)
                 throw new FileSystemException(target.toString());
+            if (x.errno() == ENOTDIR) {
+                x.setError(ENOENT);
+            }
```

Comment:
This is the mkdir to create the target so I don't think we should change this as it would be very confusing to see NoSuchFileException here.

Reply:
Reverted in 0908e75.